qemu/tests/qemu_guest_agent.py

```diff
@@ -2530,15 +2530,66 @@ def gagent_check_fsinfo(self, test, params, env):
-        2) check disk name.
-        3) check disk's serial number.
+        1) Check filesystem usage statistics
+        2) check file system type of every mount point.
+        3) check disk name.
+        4) check disk's serial number.
 
         :param test: kvm test object
         :param params: Dictionary with the test parameters
         :param env: Dictionary with test environment.
 
         """
+        def qga_guest_diskusage(mountpoint):
+            """
+            Send cmd in guest to get disk usage.
+            :param mountpoint: the mountpoint of filesystem
+            """
+            cmd_get_diskusage = params["cmd_get_disk_usage"] % mountpoint
+            disk_usage_guest = session.cmd(cmd_get_diskusage).strip().split()
+            disk_total_guest = int(disk_usage_guest[0])
+            if params["os_type"] == "windows":
+                # Just can get total and freespace disk usage from windows.
+                disk_freespace_guest = int(disk_usage_guest[1])
+                disk_used_guest = int(disk_total_guest - disk_freespace_guest)
+            else:
+                disk_used_guest = int(disk_usage_guest[1])
+            disk_total_qga = int(fs["total-bytes"])
+            disk_used_qga = int(fs["used-bytes"])
+            diff_total_qga_guest = abs(disk_total_guest - disk_total_qga)
+            diff_used_qga_guest = abs(disk_used_guest - disk_used_qga)
+            return (diff_total_qga_guest, diff_used_qga_guest)
+
+        def check_usage_qga_guest(mount_point):
+            """
+            Contrast disk usage from guest and qga that needed
+            to call previous function 'qga_guest_diskusage'.
+            :param mountpoint: the mountpoint of filesystem
+            """
+            disk_usage_guest = qga_guest_diskusage(mount_point)
+            diff_total_qgaguest = int(disk_usage_guest[0])
+            diff_used_qgaguest = int(disk_usage_guest[1])
+            if diff_total_qgaguest != 0:
+                test.fail("File System Total bytes doesn't match." %
+                          mount_pt)
+            if diff_used_qgaguest != 0:
+                if mount_point != 'C:' or mount_point != '/':
+                    test.fail("File system used bytes doesn't match." %
+                              mount_pt)
+                else:
+                    # Disk 'C:' and '/' used space usage have a floating interval,
+                    # so set a safe value '10485760'.
+                    logging.info("Need to check the floating interval for C: or /.")
+                    if diff_used_qgaguest > 10485760:
+                        test.fail("File System floating interval is too large,"
+                                  "Something must go wrong.")
+                    else:
+                        logging.info("File system '%s' usages are within the safe "
+                                     "floating range." % mount_pt)
+                        return
+            else:
```

Comment:
Don't have this else is okay.
And why add return in L2589

Reply:
okay,delete else.
return just for terminating function,but it was for last editing. I've deleted it now. it's useless now.

Reply:
pls update the code,thanks.

Reply:
@vivianQizhu xiagao has already on vacation,so please help to review this patch thanks.